internal/npm_install/generate_build_file.ts

```diff
@@ -79,78 +79,107 @@ if (require.main === module) {
   main();
 }
 
+const compareDep = (a: Dep, b: Dep) => {
+  if(a._dir < b._dir) return -1;
+  if(a._dir > b._dir) return 1; 
+  return 0;
+}
+
+const constant = <T>(c: T) => () => c
+
+async function exists(p: string) {
+  return fs.access(p, constants.F_OK | constants.W_OK)
+  .then(constant(true), constant(false));
+}
+
+// Avoid duplicate mkdir call when mkdir runs parallel.
+const mkdirPromiseMap = new Map<string, Promise<void>>();
+
 /**
  * Create a new directory and any necessary subdirectories
  * if they do not exist.
  */
-function mkdirp(p: string) {
-  if (!fs.existsSync(p)) {
-    mkdirp(path.dirname(p));
-    fs.mkdirSync(p);
+async function mkdirp(p: string) {
+  let promise = mkdirPromiseMap.get(p);
+  if (!promise) {
+    promise = (async () => {
+      if (await exists(p)) return;
```

Comment:
How about switching back to `if (!async exists(p))`, or just add the `{...}` and put the return on a newline. More style issues that should be auto-enforced :/

Reply:
Or I guess we do this single-line `if (x) return;` eslewhere, I think the indentation is just off here? Just fixing the indentation is fine as well...